pkg/sqlutil/hook_test.go

```diff
@@ -0,0 +1,124 @@
+package sqlutil
+
+import (
+	"context"
+	"database/sql"
+	"errors"
+	"fmt"
+	"runtime"
+	"testing"
+	"time"
+
+	"github.com/jmoiron/sqlx"
+	"github.com/stretchr/testify/assert"
+	"github.com/stretchr/testify/require"
+	"go.uber.org/zap/zapcore"
+
+	"github.com/smartcontractkit/chainlink-common/pkg/logger"
+	"github.com/smartcontractkit/chainlink-common/pkg/utils/tests"
+)
+
+const (
+	getDur = 10 * time.Millisecond
+	selDur = 200 * time.Millisecond
+)
+
+func TestNewInterceptedQueryer(t *testing.T) {
+	lggr, ol := logger.TestObserved(t, zapcore.InfoLevel)
+	var q Queryer = &queryer{}
+	var sentinelErr = errors.New("intercepted error")
+	const fakeError = "fake warning"
+	q = NewHookingQueryer(q, lggr, TimeoutHook(selDur/2), noopHook, MonitorHook(func() bool { return true }), noopHook, func(ctx context.Context, lggr logger.Logger, do func(context.Context) error, query string, args ...any) error {
```

Comment:
does this test validate the order of the hook executions? if so, can you tell me where that is happening?

Reply:
It does insofar as it wouldn't work the same if they were reversed, but I'll try to find a way to make it more clear.